sdk/metric/doc.go

```diff
@@ -63,6 +63,23 @@
 //     issues for databases that cannot handle high cardinality.
 //   - A too low of a limit causes loss of attribute detail as more data falls into overflow.
 //
+// # Ordering and Collection Guarantees
+//
+// For performance reasons, the SDK does not guarantee that the order in which
+// synchronous measurements are made to the SDK is reflected in the collected
+// metric data. This means that even when a single goroutine makes sequential
+// synchronous measurements, it is possible for a later measurement to be
+// included in the collected metric data when the earlier measurement is not.
+// This applies to measurements made to different instruments, or to different
+// attribute sets on the same instrument.
```

Comment:
Is it implied that measurements to the same instrument and attribute set will have their order preserved?

Reply:
Yeah, I think making that explicit may be helpful.